src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -192,29 +194,65 @@ QStringList mp::QemuPlatformDetail::vm_platform_args(const VirtualMachineDescrip
          << QString::fromStdString(fmt::format("tap,ifname={},script=no,downscript=no,model=virtio-net-pci,mac={}",
                                                tap_device_name, vm_desc.default_mac_address));
 
+    auto bridge_helper_path = QCoreApplication::applicationDirPath() + "/bridge_helper";
+
+    for (const auto& extra_interface : vm_desc.extra_interfaces)
+    {
+        opts << "-nic"
+             << QString::fromStdString(fmt::format("bridge,br={},model=virtio-net-pci,mac={},helper={}",
+                                                   extra_interface.id,
+                                                   extra_interface.mac_address,
+                                                   bridge_helper_path));
+    }
+
     return opts;
 }
 
-void mp::QemuPlatformDetail::add_network_interface(VirtualMachineDescription& desc,
-                                                   const NetworkInterface& extra_interface)
+std::vector<mp::NetworkInterfaceInfo> mp::QemuPlatformDetail::networks() const
 {
-    // TODO: Do not uncomment the following line when implementing bridging in Linux QEMU. Since implementing it would
-    // yield the same exact implementation than in macOS, we need to move the code out of here, and put it only once
-    // in src/platform/backends/qemu/qemu_virtual_machine.[h|cpp].
-    // desc.extra_interfaces.push_back(net);
+    const auto& platform_ifs_info = MP_PLATFORM.get_network_interfaces_info();
+
+    std::vector<NetworkInterfaceInfo> networks;
+
+    for (const auto& ifs_info : platform_ifs_info)
+    {
+        const auto& info = ifs_info.second;
+        const auto& type = info.type;
+
+        if (type == "bridge" || type == "ethernet")
+        {
+            networks.push_back(info);
+        }
+    }
+
+    for (auto& net : networks)
+    {
+        if (net.type == "ethernet")
+        {
+            net.needs_authorization = (find_bridge_with(networks, net.id) == networks.cend());
+        }
+    }
 
-    // For the time being, just complain:
-    throw NotImplementedOnThisBackendException("add interfaces");
+    return networks;
 }
 
 std::vector<mp::NetworkInterfaceInfo>::const_iterator mp::QemuPlatformDetail::find_bridge_with(
     const std::vector<mp::NetworkInterfaceInfo>& networks,
     const std::string& member_network) const
 {
-    return networks.cend();
+    return std::find_if(std::cbegin(networks),
```

Comment:
There is duplication happening here now with the duplicate being found in `base_virtual_machine_factory.cpp`.

What if you pulled out 
```

    for (auto& net : networks)
    {
        if (net.type == "ethernet")
        {
            net.needs_authorization = (find_bridge_with(networks, net.id) == networks.cend());
        }
    }
```

from `networks()` above and moved it to `networks()` in `qemu_virtual_machine_factory`. That way you take advantage of the inherited `find_bridge_with()` method.

Reply:
The problem with doing that is that we need to ask for authorization only on Linux. If we move it, then we'll be setting the need for authorization on macOS also. I thought a lot about this when implementing, and concluded that this is the easiest way of doing it. But please let me know if I am missing something.

Reply:
Ah, ok. I forgot about the private side. Yes, then this makes sense. At least there isn't much for duplicated logic, just a loop through a list.